crates/re_types/src/datatypes/vec2d.rs

```diff
@@ -95,9 +95,12 @@ impl ::re_types_core::Loggable for Vec2D {
                 use arrow2::{buffer::Buffer, offset::OffsetsBuffer};
                 let data0_inner_data: Vec<_> = data0
                     .iter()
-                    .flatten()
-                    .flatten()
-                    .cloned()
+                    .flat_map(|v| match v {
+                        Some(v) => itertools::Either::Left(v.iter().cloned()),
+                        None => itertools::Either::Right(
+                            std::iter::repeat(Default::default()).take(2usize),
+                        ),
+                    })
```

Comment:
it's very unfortunate that this affects things like vec2d where we know (..?) that elements are not nullable.

Reply:
Yeah, we would likely benefit from a `to_arrow` optimization in the non-opt case... however that would start much higher up with a simplified api signature that didn't consume opt types similar to what we do on the deserialization side